tasks/modules.py

```diff
@@ -366,19 +382,25 @@ def for_each(ctx: Context, cmd: str, skip_untagged: bool = False):
 
 
 @task
-def validate(_: Context, fail_fast: bool = False):
+def validate(_: Context):
     """
     Test if every module was properly added in the DEFAULT_MODULES list.
     """
-    missing_modules = []
-    for module in DEFAULT_MODULES.values():
-        for dependency in module.dependencies:
-            if dependency not in DEFAULT_MODULES:
-                if fail_fast:
-                    raise Exit(f"{color_message('ERROR', 'red')}: {module.path} depends on missing {dependency}")
-                missing_modules.append((module, dependency))
+    missing_modules: list[str] = []
+
+    # Find all go.mod files and make sure they are registered in DEFAULT_MODULES
+    for root, dirs, files in os.walk("."):
+        # Ignore the files that could be stored in the Python tests
+        dirs[:] = [d for d in dirs if root + '/' + d != "./tasks"]
```

Comment:
Is it equivalent to this ?
```suggestion
        dirs.remove(root + '/' + d)
```

Reply:
`d` is part of the list, you can't use it that way

Reply:
but `dirs.remove("./tasks")` would?

Reply:
no it won't because I need `root`. And I can't just remove `tasks` because I want to check the full path

Reply:
Indeed it should be the same as `if root == '.': dirs.remove('tasks')` isn't ?

Reply:
what do you think of https://github.com/DataDog/datadog-agent/pull/26931/commits/778da6148c98a8d2951b45f630f57f0886ea2d91?